packages/build/tests/blobs_upload/tests.js

```diff
@@ -0,0 +1,89 @@
+import { mkdtemp } from 'node:fs/promises'
+import { tmpdir } from 'node:os'
+import { join } from 'node:path'
+
+import { BlobsServer, getDeployStore } from '@netlify/blobs'
+import { Fixture, normalizeOutput } from '@netlify/testing'
+import test from 'ava'
+import getPort from 'get-port'
+
+const TOKEN = 'test'
+
+test.beforeEach(async (t) => {
+  const port = await getPort()
+  t.context.blobRequestCount = { set: 0, get: 0 }
+
+  t.context.blobServer = new BlobsServer({
+    port,
+    token: TOKEN,
+    directory: await mkdtemp(join(tmpdir(), 'blobs-upload-plugin-')),
+    onRequest: ({ type }) => {
+      t.context.blobRequestCount[type] = (t.context.blobRequestCount[type] || 0) + 1
+    },
+  })
+
+  await t.context.blobServer.start()
+
+  process.env.NETLIFY_BLOBS_CONTEXT = Buffer.from(
+    JSON.stringify({
+      edgeURL: `http://localhost:${port}`,
+      token: TOKEN, // This is for tests, during build this is passed as argument
```

Comment:
I had to provide this here because providing `token` flag in the fixtures made the tests fail. I think it's because it tries to communicate with the API. I tried `runBuildServer` in the fixtures but didn't work. I'd love to test it properly with the token passed as argument rather than using this workaround for tests. Any pointers are appreciated!

Reply:
Are you saying you tried to pass `token` as an option to `getDeployStore` and that failed? That's strange. Happy to pair on this.

Reply:
No, sorry. I tried passing it to `withFlags` on the tests so it gets passed automatically, but it failed. So in https://github.com/netlify/build/pull/5404/files#diff-05505b9d08762ab76b5e42a735b1b91aeb802a3ddd475f25002474e5e593c763R56-R59

I tried this and tests stalled
```
const output = await new Fixture('./fixtures/src_with_blobs')
    .withFlags({ deployId: 'abc123', siteId: 'test', token: TOKEN })
    .runWithBuild()
```